### PR TITLE
[docs] Improve and update Fonts guide

### PR DESCRIPTION
**Pull request by amandeepmittal**
# Why

This PR improves the Fonts guide under the Develop section. The changes follow-up #28804 and is in continuation to improve the guides under Develop section.

Closes ENG-12276

# How

- Restructure the guide to use a more "how-to" guide approach for using custom and Google Fonts
- Put all the additional/good-to-know information under Additional information section
- Provide examples at each step and update the existing ones accordingly
- Remove the outdated React Class component section and example for `Font.loadAsync`
- Define what's a PostScript name is and how to reference a font name in the project when using a local file or Google Fonts
- Add benefits and limitations to each approach (Embedding Local file with config plugin and using `useFonts` hook)
- Update internal links in Icons guide
- Update Redirects after changing the section name in Fonts guide
- Improve and update the example under the Usage section in Font API reference #29905. Changes applied to unversioned and SDK 51.

 

# Test Plan

Run docs locally and visit: http://localhost:3002/develop/user-interface/fonts or See Preview: /develop/user-interface/fonts/.

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/develop/user-interface/fonts.mdx [hunk 1]
```diff
@@ -59,323 +77,349 @@ Install [`expo-font`](/versions/latest/sdk/font/#installation) library and add t
...
+### With `expo-font` config plugin
 
-## Use a Google Font
+> **Note:** Embedding a Google Font using `expo-font` config plugin has same benefits and limitations as embedding a custom font on your own. See [using a local font file with `expo-font` config plugin](#with-expo-font-config-plugin) for more information.
 
-Expo has first-class support for all fonts listed in [Google Fonts](https://fonts.google.com/). To use one of these, check out the [`expo-google-fonts`](https://github.com/expo/google-fonts) package. With these packages, you can quickly integrate any font or font variants.
+<Step label="1">
 
-For example, to use Inter font you can install the [`@expo-google-fonts/inter`](https://www.npmjs.com/package/@expo-google-fonts/inter) package with the command below.
+Install the font package. For example, to use Inter-Black font, install the [`@expo-google-fonts/inter`](https://www.npmjs.com/package/@expo-google-fonts/inter) package with the command below.
```
Comment (Simek) [inline docs/pages/develop/user-interface/fonts.mdx:208]:
```suggestion
Install the font package. For example, to use Inter Black font, install the [`@expo-google-fonts/inter`](https://www.npmjs.com/package/@expo-google-fonts/inter) package with the command below.
```

Naming nit.

File: docs/pages/develop/user-interface/fonts.mdx [hunk 2]
```diff
@@ -59,323 +77,349 @@ Install [`expo-font`](/versions/latest/sdk/font/#installation) library and add t
...
-import React from 'react';
-import { View, Text, StyleSheet } from 'react-native';
-import { useFonts, Inter_900Black } from '@expo-google-fonts/inter';
+Each google Fonts package provides the `useFonts` hook to load the fonts asynchronously. This hook keeps track of the loading state and loads the font when an app is initialized. The font package also imports the font file so you don't have to explicitly import it.
 
-export default function App() {
-  let [fontsLoaded, fontError] = useFonts({
+<Step label="1">
+
+Instal the Google Fonts package, `expo-font` and `expo-splash-screen` libraries.
```
Comment (Simek) [inline docs/pages/develop/user-interface/fonts.mdx:269]:
```suggestion
Install the Google Fonts package, `expo-font` and `expo-splash-screen` libraries.
```

Typo fix.

File: docs/pages/develop/user-interface/fonts.mdx [hunk 3]
```diff
@@ -59,323 +77,349 @@ Install [`expo-font`](/versions/latest/sdk/font/#installation) library and add t
...
+
+</Step>
+
+<Step label="2">
+
+After installing the font package, map the font using the `useFonts` hook in a top level component such as the root layout (**app/layout.tsx**) file in your project:
+
+```tsx app/_layout.tsx
+// Rest of the import statements
+/* @info Import <CODE>useFonts</CODE> hook from <CODEexpo-font</CODE>. */ import { useFonts } from 'expo-font'; /* @end */
```
Comment (Simek) [inline docs/pages/develop/user-interface/fonts.mdx:283]:
```suggestion
/* @info Import <CODE>useFonts</CODE> hook from <CODE>expo-font</CODE>. */ import { useFonts } from 'expo-font'; /* @end */
```

Typo fix.

**Review by Simek — APPROVED** [review]
Looks good! 👍 Left just few nits here and there.

File: docs/pages/guides/icons.mdx [hunk 4]
```diff
@@ -202,13 +205,15 @@ export default function App() {
...
-  },
+container: {
+flex: 1,
+justifyContent: 'center',
+alignItems: 'center',
+},
 });
-```
+
+````
```
[the remark this answers is not in the document]
Comment (amandeepmittal) [reply docs/pages/guides/icons.mdx:215]: Yep, fixing it.

**Pull request merged**